File: client/src/models/Meeting.ts
import type { MeetingStatus } from "./enums/MeetingStatus";

export type MeetingParticipant = {
  id: string;
  firstName: string;
  lastName: string;
};

export type MeetingOwner = MeetingParticipant;

export type Meeting = {
  id: number;

  startDateTime: string;
  endDateTime: string;

  summary: string | null;
  description: string | null;
  location: string | null;

  maxParticipants: number | null;
  status: MeetingStatus;

  freeTimeId: number | null;
  owner: MeetingOwner;

  participants: MeetingParticipant[];
  activityIds: number[] | null;
};

File: server/app/src/main/java/lt/gathertime/server/model/Meeting.java
package lt.gathertime.server.model;

import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lt.gathertime.server.model.enums.MeetingStatus;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Meeting {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "summary")
    private String summary;

    @Column(name = "description")
    private String description;

    @Column(name = "location")
    private String location;

    @Column(name = "max_participants")
    private Integer maxParticipants;

    @Enumerated(EnumType.STRING)
    private MeetingStatus status;

    @ManyToMany
    @JoinTable(
        name = "meeting_participants",
        joinColumns = @JoinColumn(name = "user_id"),
        inverseJoinColumns = @JoinColumn(name = "meeting_id")
    )
    private List<User> meetingParticipants;

    @ManyToMany
    @JoinTable(
        name = "meeting_activities",
        joinColumns = @JoinColumn(name = "meeting_id"),
        inverseJoinColumns = @JoinColumn(name = "activity_id")
    )
    private List<Activity> meetingActivities;
}